js/src/Dapps/dapps.js

```diff
@@ -45,50 +46,82 @@ class Dapps extends Component {
     this.store.loadAllApps();
   }
 
+  handlePin = (appId) => {
+    if (this.store.displayApps[appId].pinned) {
+      this.store.unpinApp(appId);
+    } else {
+      this.store.pinApp(appId);
+    }
+  }
+
+  renderSection = (title, apps) => (
+    apps && apps.length > 0 && <div>
+      <Header as='h4' className={ styles.sectionTitle }>
+        {title}
+      </Header>
+      <div className={ styles.dapps }>
+        {
+          apps.map((app, index) => (
+            <DappCard
+              app={ app }
+              pinned={ this.store.displayApps[app.id] && this.store.displayApps[app.id].pinned }
+              availability={ this.props.availability }
+              className={ styles.dapp }
+              key={ `${index}_${app.id}` }
+              onPin={ this.handlePin }
+            />
+          ))
+        }
+      </div></div>
+  )
+
   render () {
-    const { availability } = this.props;
-    const applications = [].concat(this.store.visibleLocal, this.store.visibleViews, this.store.visibleBuiltin, this.store.visibleNetwork);
+    const pinned = this.store.pinnedApps; // Pinned apps
+    const unpinnedVisible = this.store.visibleApps.filter(app => // Visible apps that are not pinned
```

Comment:
This should probably be in the store? (Can be computed?)

Reply:
Yes definitely.